simulations/csh-benchmarks/R/eval-functions.R

```diff
@@ -0,0 +1,231 @@
+################################################################################
+## Evaluator Functions
+################################################################################
+
+#' Extract correlation matrix from model fits
+#'
+#' @description Extracts the repeated measures' covariance matrix estimated by
+#'   the various fitting procedures.
+#'
+#' @param fit An object returned by either the mrmm_wrapper_fun(),
+#'   glmmTMB_wrapper_fun() or nlme_wrapper_fun().
+#'
+#' @return The fit argument's estimated repeated measures covariance matrix.
+get_covar_mat <- function(fit) {
+
+  ## extract the covariance matrix from the fit object
+  if (class(fit)[1] == "gls") {
+    covar_mat <- nlme::getVarCov(fit)
+  } else if (class(fit)[1] == "glmmTMB") {
+    corr_fit_attributes <- attributes(VarCorr(fit)[["cond"]]$participant)
+    corr_mat <- corr_fit_attributes$correlation
+    out_stddev_mat <- outer(corr_fit_attributes$stddev, corr_fit_attributes$stddev)
+    covar_mat <- corr_mat * out_stddev_mat
+  } else if (class(fit)[1] == "mmrm") {
+    covar_mat <- fit$cov
+  } else if (class(fit)[1] == "data.frame") { # SAS PROC MIXED
+    cor_est_mat <- matrix(fit$Estimate[11], nrow = 10, ncol = 10)
+    diag(cor_est_mat) <- 1
+    std_devs <- sqrt(fit$Estimate[1:10])
+    unweighted_covar_mat <- outer(std_devs, std_devs)
+    covar_mat <- cor_est_mat * unweighted_covar_mat
```

Comment:
this seems worth to factor out in a function and test it separately

Reply:
in particular we need to absolutely sure that the order of the time points matches the order we have in R

Reply:
Good idea. Will do!